data_scripts/structure_data.py

```diff
@@ -27,17 +27,33 @@ def assign_link_lengths(network_fp: str, node_pairs_dict: dict, constant_weight:
 
     return response_dict
 
+def assign_core_nodes(core_nodes_fp: str):
+    """
+    List a core nodes of network topology. 
+
+    :param core_nodes_fp: File path that contains core nodes  information.
+    :return: A list of core nodes.
+    :rtype: list
+    """
+    response_list = []
+    with open(core_nodes_fp, 'r', encoding='utf-8') as file_obj:
+        for line in file_obj:
+            response_list.append(line.strip().split('\t')[0])
+    return response_list
+
 
-def create_network(net_name: str, base_fp: str = None, const_weight: bool = False):
+def create_network(net_name: str, base_fp: str = None, const_weight: bool = False, is_only_core_node: bool = True):
     """
     The main structure data function.
 
     :param net_name: The desired network name, used to read the data file.
     :param const_weight: Set all links' weights to one if true.
     :param base_fp: The base of the file path to read the network from.
+    :param is_only_core_node: Network includes only core nodes or not.
     :return: The network spectrum database.
     :rtype: dict
     """
+    core_nodes= []
```

Comment:
Do you want to include this in some object's properties? Like `sdn_props` or something similar? You could have access to it across multiple scripts

Reply:
It is assigned to engine_props